### DISCUSSION TITLE
javascript error: Cannot read properties of null (reading 'value')

### DISCUSSION THREAD

**Discussion by poojapatel14710** [seleniumbase/SeleniumBase#1596] [Q&A] [answered]
I am trying to get the default value from the textbox but it shows "selenium.common.exceptions.JavascriptException: Message: javascript error: Cannot read properties of null (reading 'value') ".

below picture is for HTML and Textbox :
![image](https://user-images.githubusercontent.com/76393417/199911558-3a8a5de5-f99b-4e89-b3fe-a725915469bb.png)

and my script is:
```
    def test_default_value(self):
        self.open(c2cSms_data.url)
        self.wait_for_element_visible(C2cSMS.input.format([self.index]))
        default_value = self.get_value(C2cSMS.input.format([self.index]))
        self.assert_equal(default_value, 1)
```

**Comment by mdmintz** [comment, marked as answer]
Hello. You can use ``get_text()`` to get the value of an input field assuming you have the correct selector. Eg:

```python
value = self.get_text("input")
```

**Reply by poojapatel14710** [reply]
Thank you sir for your quick response!! and it's work.

but could you please explain to me why I should get text instead of value?

**Reply by mdmintz** [reply]
In your case, ``document.querySelector()`` returned `null`, so the JS call didn't find the element. Maybe you used a different selector the first time, or the JS call wasn't smart enough.
